src/injecting/injector.rs

```diff
@@ -72,20 +93,92 @@ impl FunctionMapper for Module {
             })
             .ok_or("No malloc function in table section")?;
 
+        Ok(malloc_index)
+    }
+    fn map_function(
+        &mut self,
+        function_name: &str,
+        body_mapper: impl Fn(&mut FuncBody, usize),
+    ) -> Result<(), String> {
+        // NOTE:
+        // Indexing for local functions includes both imported and own (local)
+        // functions. So we actually need the imported function count to resolve the
+        // actual index of the given function in own functions list.
+
+        // Find the global function index
+        let global_function_index = self.get_global_function_index(function_name).unwrap();
+
+        // Count the number of imported functions listed by the module
+        let import_section_len: usize = self.get_import_section_len().unwrap();
+
+        let malloc_index = self.get_malloc_index().unwrap();
```

Comment:
You could either add a short comment here (regarding why this is complex enough that it needs to be abstracted away in a function).

I've commented about the use of `.unwrap()` below.